src/data/bool/basic.lean

```diff
@@ -225,4 +225,8 @@ by cases b; simp only [of_nat,to_nat]; exact dec_trivial
 ⟨λ Hinj Heq, ff_ne_tt (Hinj Heq),
   λ H x y hxy, by { cases x; cases y, exacts [rfl, (H hxy).elim, (H hxy.symm).elim, rfl] }⟩
 
+/-- **Kaminski's Equation** -/
+theorem apply_apply_apply (f : bool → bool) (x : bool) : f (f (f x)) = f x :=
+by cases x; cases h₁ : f tt; cases h₂ : f ff; simp only [h₁, h₂]
```

Comment:
Does this shorter proof work in this file?
```suggestion
theorem apply_apply_apply : ∀ (f : bool → bool) (x : bool), f (f (f x)) = f x :=
dec_trivial
```

Reply:
I checked locally and it seems like it doesn't

Reply:
Neither does Patrick Johnson's `fin_cases` proof

Reply:
@eric-wieser What imports did that need? This module is a mathlib root (no imports!)

Reply:
It needs `data.fintype.basic`:
```
import data.fintype.basic

lemma foo : ∀ (f : bool → bool) {x : bool}, f (f (f x)) = f x :=
dec_trivial
```
as it uses `pi.fintype`.

Reply:
Surely this would make for a nice (if niche) `simp` theorem?

Reply:
It's not valid because the head symbol is not a constant, so lean can infer `f = id` and get stuck in a loop